### PR TITLE
[kubernetes] Support --config to mount buildkit.toml and -driver-opt=qemu.install=true,qemu.image=tonistiigi/binfmt:latest for qemu installing

### PR DESCRIPTION
**Pull request by morlay**
fix https://github.com/docker/buildx/issues/603

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/docker/buildx
 
-go 1.13
+go 1.16
```
Comment (morlay) [inline go.mod:3]: `os.ReadFile` required

**Comment by morlay** [comment]
Is master branch broken? 
Not sure why I have to updates `vendor` and `go{.mod,.sum}`

**Comment by tonistiigi** [comment]
> Not sure why I have to updates vendor and go{.mod,.sum}

Because of the version change probably.

**Review by tonistiigi — APPROVED** [review]
SGTM 👍 

PTAL @AkihiroSuda

**Comment by morlay** [comment]
@tonistiigi I added docs to fix https://github.com/docker/buildx/issues/516 too.

Now sure the docs should be here

**Comment by morlay** [comment]
@tonistiigi I added docs to fix https://github.com/docker/buildx/issues/516 too.

Now sure the docs should be here

File: docs/reference/buildx_create.md [hunk 2]
```diff
@@ -123,6 +123,8 @@ Passes additional driver-specific options. Details for each driver:
       ```
 - `kubernetes`
     - `image=IMAGE` - Sets the container image to be used for running buildkit.
+    - `withqemu=(true|false)` - Install QEMU emulation for multi platforms support.
+    - `qemuimage=IMAGE` - Sets the QEMU emulation image. Defaults to `tonistiigi/binfmt:latest`
```
Comment (AkihiroSuda) [inline docs/reference/buildx_create.md:127]: nit: what about renaming them to `qemu.install` and `qemu.image`

File: docs/reference/buildx_create_for_multi_arch_build_in_k8s.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    docker buildx create \
+        --name=builder \
+        --platform=linux/amd64,linux/arm64 \
+        --driver=kubernetes \
+        --driver-opt=namespace=buildkit,withqemu=true
+```
+
+### Known Issues
+
+QEUM is only work well for `executing`, but for `compiling`, it wil be very slow and may be panic.
```
Comment (AkihiroSuda) [inline docs/reference/buildx_create_for_multi_arch_build_in_k8s.md:16]: seems to have a grammatical issue

**Comment by AkihiroSuda** [comment]
>  Support --config to mount buildkit.toml and -driver-opt=withqemu=true,qemuimage=tonistiigi/binfmt:latest for qemu installing #682

This PR seems mixing up two different topics.

Next time please consider splitting PRs (or splitting commits) .

File: docs/reference/buildx_create.md (hunk 2 above)
Comment (morlay) [reply docs/reference/buildx_create.md:127]: Sure. looks better.

File: docs/reference/buildx_create_for_multi_arch_build_in_k8s.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    docker buildx create \
+        --name=builder \
+        --platform=linux/amd64,linux/arm64 \
+        --driver=kubernetes \
+        --driver-opt=namespace=buildkit,withqemu=true
+```
+
+### Known Issues
+
+QEUM only work well for `executing`, but for `compiling`, it will be very slow and may be panic.
```
Comment (AkihiroSuda) [inline docs/reference/buildx_create_for_multi_arch_build_in_k8s.md:16]: - typo: QEUM

- No need to quote "executing" and "compiling"

- What do you mean by "panic"?
-

File: docs/reference/buildx_create_for_multi_arch_build_in_k8s.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+In the mode, if still want to compile binaries in docker.
+please use `FROM --platform=${BUILDPLATFORM}` to disable QEUM for compiling stage.
```
Comment (AkihiroSuda) [inline docs/reference/buildx_create_for_multi_arch_build_in_k8s.md:19]: I don't understand how `--platform=${BUILDPLATFORM}` relates to docker here

File: docs/reference/buildx_create_for_multi_arch_build_in_k8s.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+The build time may be longer. 
+Even `FROM --platform=${BUILDPLATFORM}` defined, all stages will build for each arch.
+
+However, it is totally native.
+Projects, witch needs to build on native system, will be happy with the mode.
+
+## Tips 
+
+Once `buildx create` in k8s, the created deployments will not be removed until `buildx rm` called.
+So we could set `RUN --mount=type=cache` for sharing common caches for different projects.
```
Comment (AkihiroSuda) [inline docs/reference/buildx_create_for_multi_arch_build_in_k8s.md:71]: It should be noted that `--mount=type=cache` is not shared across multiple nodes in the cluster.

File: docs/reference/buildx_create_for_multi_arch_build_in_k8s.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+However, For nodejs user.
+Don't shared `npm` or `yarn` global caches (`pnpm` will be better), restoring caches may be slower than reinstalling.
```
Comment (AkihiroSuda) [inline docs/reference/buildx_create_for_multi_arch_build_in_k8s.md:74]: This sentence should be in a separate document in a separate PR, as this is not specific to Kubernetes driver.

File: docs/reference/buildx_create_for_multi_arch_build_in_k8s.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+## With QEMU emulation
+
+```console
+$ KUBECONFIG=${KUBECONFIG} \
+    docker buildx create \
+        --name=builder \
+        --platform=linux/amd64,linux/arm64 \
+        --driver=kubernetes \
+        --driver-opt=namespace=buildkit,withqemu=true
```
Comment (AkihiroSuda) [inline docs/reference/buildx_create_for_multi_arch_build_in_k8s.md:11]: s/withqemu/qemu.install/

File: docs/reference/buildx_create_for_multi_arch_build_in_k8s.md (hunk 4 above)
Comment (morlay) [reply docs/reference/buildx_create_for_multi_arch_build_in_k8s.md:16]: > Slowness is not specific to compilation

Slowness for executing could be acceptable, but for compiling, obviously slow.

> What do you mean by "panic"?

throw some error like `qemu: uncaught target signal 11 (Segmentation fault) - core dumped`
https://github.com/docker/buildx/issues/516#issuecomment-781685997

Maybe rename to `segmentation fault`?

File: docs/reference/buildx_create_for_multi_arch_build_in_k8s.md (hunk 5 above)
Comment (morlay) [reply docs/reference/buildx_create_for_multi_arch_build_in_k8s.md:19]: Just like example below for golang.

When `--platform=${BUILDPLATFORM}` here, QEMU will not be used in the `builder` stage

File: docs/reference/buildx_create_for_multi_arch_build_in_k8s.md (hunk 7 above)
Comment (morlay) [reply docs/reference/buildx_create_for_multi_arch_build_in_k8s.md:74]: Not sure where to put.

In factor, this docs both cover `kubernetes` and `docker-container` driver.

**Comment by morlay** [comment]
@AkihiroSuda I removed the doc from this PR.

may just put in issues or discussions

**Comment by morlay** [comment]
mv to  https://github.com/docker/buildx/discussions/685

**Review by AkihiroSuda — APPROVED** [review]
We should have CI, but can be added later

**Comment by morlay** [comment]
Yes. Could setup k3s for kubernetes. 
but i have no idea how to the testing scripts though go or just prue shells

```
      - name: setup k3s
        run: |
          curl -sfL https://get.k3s.io | K3S_KUBECONFIG_MODE="644" sh -s -
          kubectl version
      - run: #scrips
        env:
          KUBECONFIG: /etc/rancher/k3s/k3s.yaml
```

**Comment by onmomo** [comment]
This great feature is now merged for quite some time but is not yet included in the latest buildx releases. Is this on purpose?
